utils/context.py

```diff
@@ -90,7 +90,7 @@ def ibridges_configuration(self) -> json_config.JsonConfig:
             Configuration instance.
 
         """
-        if self._ibridges_configuration is None:
+        if self._ibridges_configuration is None or self._ibridges_configuration.config is {}:
```

Comment:
The config file loading is intended only to be called the one time, when the `_ibridges_configuration` has never been set.  If it is re-set like this, a new object will be created and previous pointers to the old object will need to be updated to this new object _everywhere_ it is used.  This is very undesirable!

In the context of Issue #82, I have this alternate implemention that should work better:

```python
        if self._ibridges_configuration is None:
            if not self.ibridges_conf_file:
                self.ibridges_conf_file = DEFAULT_IBRIDGES_CONF_FILE
            filepath = path.LocalPath(self.ibridges_conf_file).expanduser()
            if not filepath.parent.is_dir():
                filepath.parent.mkdir()
            if not filepath.is_file():
                filepath.write_text(json.dumps(IBRIDGES_CONF_TEMPLATE))
            self._ibridges_configuration = json_config.JSONConfig(filepath)
        elif self.ibridges_conf_file != self._ibridges_configuration.filepath:
            self._ibridges_configuration.reset()
            self._ibridges_configuration.filepath = self.ibridges_conf_file
```

The new `filepath` and empty `config` dictionary will then cause the `JsonConfig` to load the configuration from the new file within the original instance.

Incidentally, this resolves an esoteric bug where the iBridges configuration is wiped and recreated when connecting after a 'Close session'.